### PR TITLE
Replace k8s/install/gce with ref/pub-cloud/gce page

### PR DESCRIPTION
**Pull request by tmjd**
The page in its reduced form is probably better suited to be in our
reference/public-cloud location.
The redirect needs to be fixed, I'm hoping to get some help with that.

Addresses #715
Alternate to #727 
Mostly copied from #388

### CODE REVIEW

**Comment by tmjd** [comment]
The way the latest redirects are updated will be incompatible with the redirect I've tried to include.

File: master/reference/public-cloud/gce.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+| Description      | Protocol | Port Range |
+|:-----------------|:---------|:-----------|
+| BGP              | TCP      | 179        |
+| \*IPIP           | 4        | all        |
+
+>\*IPIP: This rule is required only when using Calico with IPIP encapsulation.
+Keep reading for information on when IPIP is required in GCE.
+
+#### Routing Traffic
```
Comment (caseydavenport) [inline master/reference/public-cloud/gce.md]: There are two options here: using GCE cloud routes, or using IPIP.

We should talk about both.

**Comment by tmjd** [comment]
I added mention of GCE cloud routes and there is already #414 for adding specifics on how to set that up.
I also added a link from the prior GCE in the k8s section to the GCE in the reference section.

File: master/reference/public-cloud/gce.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+To use Calico in GCE, either [ipip encapsulation][IPIP] and NAT outgoing must
```
Comment (caseydavenport) [inline master/reference/public-cloud/gce.md]: I think we should split these up into two subsections of "Routing traffic"

and then we should make the answer to this question more apparent in this section:

> This rule is required only when using Calico with IPIP encapsulation. Keep reading for information on when IPIP is required in GCE

Comment (caseydavenport) [inline master/reference/public-cloud/gce.md]: We can add more details on using GCE cloud routes later.

File: master/reference/public-cloud/gce.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+#### Routing Traffic
+
+One of the following options must be utilized when using Calico in
+GCE to ensure container traffic is allowed by the GCE network fabric.
+
+##### IP-in-IP encapsulation
+
+Container traffic routing can be enabled by setting [IP-in-IP encapsulation][IPIP]
+and NAT outgoing on the configured Calico IP pools.  Setting these ensures
+that all traffic to and from containers will be sent with IP addresses that
```
Comment (tmjd) [inline master/reference/public-cloud/gce.md]: I'm not sure about this `Setting these ensures ....` sentence, it might be TMI. :grinning:

**Comment by tmjd** [comment]
@caseydavenport, @djosborne PTAL

File: master/reference/public-cloud/gce.md (hunk 3 above)
Comment (caseydavenport) [inline master/reference/public-cloud/gce.md]: Yeah, I'd err on the side of removing explanatory stuff and keeping it focused on "What do you need to do to make it work"

File: master/reference/public-cloud/gce.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+are allowed by the GCE network fabric.
+
+See the [IP pool configuration reference][IPPool]
+for information on how to configure Calico IP pools.
+
+##### GCE cloud routes
+
+Traffic routing in GCE can be achieved by utilizing GCE cloud routes.
+Kubernetes GCE cloud provider integration simplifies route configuration
+by enabling Kubernetes to handle creating routes.
```
Comment (caseydavenport) [inline master/reference/public-cloud/gce.md]: I'd just add a but of context saying adding ".. and running Calico in policy-only mode"

**Comment by tmjd** [comment]
These changes need to be applied to v2.2 before merging.  Are we there? :slightly_smiling_face:

File: _data/master/navbars/getting-started.yml [hunk 5]
```diff
@@ -27,8 +27,6 @@ toc:
       path: /getting-started/kubernetes/installation/integration
     - title: AWS
       path: /getting-started/kubernetes/installation/aws
-    - title: GCE
-      path: /getting-started/kubernetes/installation/gce
```
Comment (djosborne) [inline _data/master/navbars/getting-started.yml]: I think this needs to stay?

File: master/getting-started/kubernetes/installation/gce.md [hunk 6]
```diff
@@ -2,140 +2,5 @@
...
-## Next Steps
-
-You should now have a fully functioning Kubernetes cluster using Calico for networking.  You're ready to use your cluster.
-
-We recommend you try using [Calico for Kubernetes NetworkPolicy]({{site.baseurl}}/{{page.version}}/getting-started/kubernetes/tutorials/simple-policy).
-
-[cni-plugin]: https://github.com/projectcalico/cni-plugin
-[coreos-gce]: https://coreos.com/os/docs/latest/booting-on-google-compute-engine.html
-[gcloud-instructions]: https://cloud.google.com/sdk/
+See [this page]({{site.baseurl}}/{{page.version}}/reference/public-cloud/gce)
```
Comment (caseydavenport) [inline master/getting-started/kubernetes/installation/gce.md]: Do we have an idea what the future of this page will be? Will it eventually just be removed, or can we turn it into something like this http://docs.projectcalico.org/v2.1/getting-started/kubernetes/installation/aws?

I think leaving it as a link is OK for now, but ideally we follow up quickly-ish with something similar to the AWS doc.  WDYT?

**Comment by caseydavenport** [comment]
LGTM - we can deal with follow on actions for the getting started GCE doc in the near future.

File: master/getting-started/kubernetes/installation/gce.md (hunk 6 above)
Comment (tmjd) [inline master/getting-started/kubernetes/installation/gce.md]: From my talking with Turk I believe we expect those pages to go away as much as possible with only stuff that would be directly needed for k8s.  The reason the page was not dropped altogether was to have a way to redirect to the new page.  (Using a redirect_from would not mesh with our current `latest` redirection.)

**Comment by tmjd** [comment]
@djosborne I added that link back in and applied the changes to v2.2, can you merged this please?

File: master/getting-started/kubernetes/installation/gce.md (hunk 6 above)
Comment (caseydavenport) [inline master/getting-started/kubernetes/installation/gce.md]: I think I disagree with @djosborne about wanting those pages to go away as much as possible, but that's a discussion we can have later :)

**Pull request merged**